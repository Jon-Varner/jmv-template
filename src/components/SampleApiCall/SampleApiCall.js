import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SampleApiCall extends Component {
  componentDidMount() {
    this.props.userFetched();
  }

  render() {
    const user = this.props.user;
    let id = 0;
    let name = 'Unavailable';

    if (user.hasOwnProperty('id')) {
      id = user.id;
      name = user.name;
    }

    return (
      <p>
        API returned User ID: {id} and Name: {name}.
      </p>
    );
  }
}

SampleApiCall.propTypes = {
  user: PropTypes.object,
  userFetched: PropTypes.func.isRequired
};

export default SampleApiCall;
