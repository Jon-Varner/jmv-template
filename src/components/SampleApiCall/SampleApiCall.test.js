import React from 'react';
import { mount } from 'enzyme';

import SampleApiCall from './SampleApiCall';

describe('SampleApiCall renders props correctly', () => {
  let props;
  let mountedSampleApiCall;

  const sampleForm = () => {
    if (!mountedSampleApiCall) {
      mountedSampleApiCall = mount(<SampleApiCall {...props} />);
    }
    return mountedSampleApiCall;
  };

  beforeEach(() => {
    props = {
      user: {
        id: undefined,
        name: undefined
      },
      userFetched: jest.fn()
    };

    mountedSampleApiCall = undefined;
  });

  it('always renders a user id and name', () => {
    const para = sampleForm().find('p');
    expect(para.text()).toBe('API returned User ID: 0 and Name: Unavailable.');
  });

  it('renders user id and name from props', () => {
    props = {
      user: {
        id: 1,
        name: 'first name'
      },
      userFetched: jest.fn()
    };

    const para = sampleForm().find('p');
    expect(para.text()).toBe('API returned User ID: 1 and Name: first name.');
  });
});
