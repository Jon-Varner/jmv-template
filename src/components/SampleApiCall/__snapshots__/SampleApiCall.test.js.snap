// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <p
    class="api-result"
  >
    API returned Person ID: 0 and Name: Initial User Name.
  </p>
</DocumentFragment>
`;
